src/transform/inverse.rs

```diff
@@ -94,12 +100,16 @@ pub fn av1_iadst4(input: &[i32], output: &mut [i32], _range: usize) {
 }
 
 pub fn av1_iidentity4(input: &[i32], output: &mut [i32], _range: usize) {
-  for i in 0..4 {
-    output[i] = round_shift(SQRT2 * input[i], 12);
-  }
+  output[..4]
+    .iter_mut()
+    .zip(input[..4].iter())
+    .for_each(|(outp, inp)| *outp = round_shift(SQRT2 * *inp, 12));
```

Comment:
I would rather you use iterators and a for loop for the identity transforms. It's up to you.

Reply:
I personally think the fully functional version looks cleaner. But it's a matter of personal preference. :)